Cargo.toml

```diff
@@ -12,6 +12,10 @@ autoexamples = false
 exclude =  ["assets"]
 resolver = "2"
 
+[features]
+u16-packetsize = []
+default = []
```

Comment:
Features are supposed to be additive, but clearly on cannot have both u32 and u16 enabled at the same time. 

So I'd like to see this not implemented as a feature, but rather as a configuration. Since the crate should be configurable anyway, this would lay out the groundwork for future improvements.

I think that extending the `NetworkSetting`'s `max_packet_length` to be an enum like this:

```rust
enum MaxPacketLength {
    U32(pub u32),
    U16(pub u16),
}
```

Such a change would allow to not have the issue where for example another crate could extend this one, but then have conflicting features.

Reply:
that sounds good to me :+1:

Reply:
I may actually close this PR and redo my other WIP branch, as there is a number of things that are currently out of hands of control when implementing with this crate.  For instance, the length header that comes back can be little endian but the length you send back should be big endian ( I know it sounds crazy, but I see it sometimes  :smile:  ).  Also the size headers are not always guaranteed to either a) exist and/or be first as part of the packet.

I had adapted a way to inject my own encoding strategy here: https://github.com/benfalk/bevy_spicy_networking/commit/e883c71099e12d9affa5e6cc5955c25f8b24fc67

However; after implemented it I realized that what you really want to potentially override is the entire part of
the loop were you receive messages and act upon them.

Reply:
> length header that comes back can be little endian but the length you send back should be big endian

I am not sure why that would be the case, since `write_u32` should always be big-endian. 

> I had adapted a way to inject my own encoding strategy here: benfalk@e883c71

Yup! That looks like the correct approach for me.


-----


If you want to, you can put them both in the same PR. That would be fine from my side.